### PR TITLE
support nil logger and ActiveSupport::Logger not Logger

### LINKED ISSUE #168: undefined method `silence' for nil (NoMethodError) with latest rails main
It happened between rails 917b6b2908510fbbc8d55db6360220485324e63c and rails 608c1bfe17a2626304b6aae96a3824517fc07bb2

```
07:00:38 web.1  | /Users/dorianmariefr/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/solid_queue-0.2.2/lib/solid_queue/processes/poller.rb:14:in `with_polling_volume': undefined method `silence' for nil (NoMethodError)
07:00:38 web.1  | 
07:00:38 web.1  |           ActiveRecord::Base.logger.silence { yield }
07:00:38 web.1  |                                    ^^^^^^^^
07:00:38 web.1  | 	from /Users/dorianmariefr/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/solid_queue-0.2.2/lib/solid_queue/dispatcher.rb:31:in `dispatch_next_batch'
07:00:38 web.1  | 	from /Users/dorianmariefr/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/solid_queue-0.2.2/lib/solid_queue/dispatcher.rb:22:in `run'
07:00:38 web.1  | 	from /Users/dorianmariefr/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/solid_queue-0.2.2/lib/solid_queue/processes/runnable.rb:48:in `block (2 levels) in do_start_loop'
```

### PR DESCRIPTION
**Pull request by dorianmariecom**
FIxes #168

### CODE REVIEW

File: test/integration/processes_poller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def test_active_record_logger_active_support_logger_dev_null
+    previous_logger = ActiveRecord::Base.logger
+    ActiveRecord::Base.logger = ActiveSupport::Logger.new("/dev/null")
+    pid = fork { SolidQueue::Supervisor.start }
+    sleep 1
+  ensure
+    ActiveRecord::Base.logger = previous_logger
+    terminate_process(pid) if pid && process_exists?(pid)
+  end
```
Comment (jeremy) [inline test/integration/processes_poller.rb:24]: Unrelated test coverage?

Comment (dorianmariecom) [reply test/integration/processes_poller.rb:24]: Yeah I had a hard time trying to reproduce the bug, I don't know the library very well, so happy for pointers for better ways to do it

File: lib/solid_queue/processes/poller.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@ module Poller
 
     private
       def with_polling_volume
-        if SolidQueue.silence_polling?
+        if SolidQueue.silence_polling? && ActiveRecord::Base.logger
           ActiveRecord::Base.logger.silence { yield }
```
Comment (nbulaj) [inline lib/solid_queue/processes/poller.rb:14]: Maybe better just use safe-navigation `ActiveRecord::Base.logger&.silence` ?

Comment (dorianmariecom) [reply lib/solid_queue/processes/poller.rb:14]: Then it will not `yield`

Comment (nbulaj) [reply lib/solid_queue/processes/poller.rb:14]: But why? :thinking: 

```
irb(main):003> l = Logger.new(STDOUT)
irb(main):004> l.info { "OK" }
I, [2024-03-22T15:10:27.939697 #68513]  INFO -- : OK
=> true
irb(main):005> l&.info { "OK" }
I, [2024-03-22T15:10:31.957994 #68513]  INFO -- : OK
=> true


irb(main):008> n = nil
=> nil
irb(main):009> n&.info { "OK"  }
=> nil
```

**Pull request merged**